matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/auth/registration/RegistrationWizard.kt

```diff
@@ -39,6 +39,8 @@ interface RegistrationWizard {
 
     fun checkIfEmailHasBeenValidated(delayMillis: Long, callback: MatrixCallback<RegistrationResult>): Cancelable
 
+    fun registrationAvailable(userName: String, callback: MatrixCallback<RegistrationAvailability>): Cancelable
```

Comment:
You should create a `suspend` here, instead of using `MatrixCallback` (code in Element Android is newer on this point).

Reply:
done in https://github.com/vector-im/element-android/pull/3084